### DISCUSSION TITLE
Help Setting up Mongodb Atlas

### DISCUSSION THREAD

**Discussion by maddust** [opendatacam/opendatacam#577] [Q&A] [answered]
Hi all, 

Im trying to set my dataset in mondodb atlas (cloud) instead of having the data stored in a local instance .  

Can anyone provide some steps to set things up? 

So far i have tried changing the config.json with the bellow :

  "DATABASE": "mongo",
  "DATABASE_PARAMS": {
    "mongo": {
      "url": "mongodb+srv://<my user and password>"@opendatacam.wxt6e.mongodb.net/?retryWrites=true&w=majority",
      "persistTracker": false
    }

also installed mongosh as i read that was needed for the communication. 

im getting this msg error on the log
MongoParseError: Text record must only set `authSource` or `replicaSet`
    at QueryReqWrap.callback (/home/haroldpc/src/odc-dev/node_modules/mongodb/lib/core/uri_parser.js:99:13)

Running Develomnent brach. 

Any help how to set this up will be greatly appreciate it 

@vsaw 

thanks !

**Comment by vsaw** [comment, marked as answer]
Maybe just copy and paste error but it looks like there is a " in the middle of the connect URL. 

Generally speaking if you fix the URL also with DB login credentials it should work out fine.

**Reply by maddust** [reply]
Thank for the reply @vsaw 

Oh no regarding the ( " )  i just remove my credentials from the URL example in the post 
mongodb+srv://**credentials removed**@opendatacam.wxt6e.mongodb.net/?retryWrites=true&w=majority"

unfortunately is not working with the above string.

**Reply by vsaw** [reply]
Does the connect string work with MongoDB compass?

**Reply by maddust** [reply]
Yes the string below works in compass 
mongodb+srv://hpantaleon:<password>@opendatacam.wxt6e.mongodb.net/test

[https://imgur.com/a/uJpu5cp](url)

attached is the log when i run ODC 
[log.txt](https://github.com/opendatacam/opendatacam/files/10470811/log.txt)

**Reply by maddust** [reply]
got it working. 

i initially created a serverless database  once i try it with the server option it worked ! 

thank you for following up!

**Reply by vsaw** [reply]
Great to hear! 

FYI the ODC log contains the full config including Mongo password. So I suggest you change your Mongo PW
